### ISSUE TITLE
Research task: automate creation of GitHub App

### ISSUE THREAD

**Issue by alexellis** [alexellis/derek#70]
Can we make development and testing easier by automating the creation/config of the Github App via the GitHub API?

Ideal flow:

* User creates personal access_token
* User runs `derek --provision-app --name="Derek PR 68"`
* GitHub App created and then access token revoked?

**Comment by alexellis** [comment]
If we can do this with `curl` I think that's OK too, updating the CLI is of lower priority.

**Comment by alexellis** [comment]
@rgee0 I can't assign issues on a personal repo, but would you be willing to take this on with your automation experience it might make sense?

**Comment by alexellis** [comment]
I put this together for OpenFaaS Cloud, someone could send me a PR to add an option for Derek? https://github.com/alexellis/github-app

**Comment by Waterdrips** [comment]
@alexellis I would like to take this one

**Comment by alexellis** [comment]
I would assume you can copy my approach or vendor it from ofc-bootstrap?

**Comment by Waterdrips** [comment]
yep - looks like a nice easy one to get me started on the Derek codebase

**Comment by alexellis** [comment]
This is lower priority IMHO than the overrides and merge feature, esp. as we offer a hosted version of Derek now.

**Comment by Waterdrips** [comment]
yep, iv put it low on my list
